numpy/distutils/misc_util.py

```diff
@@ -14,9 +14,13 @@
 import distutils
 from distutils.errors import DistutilsError
 try:
-    from threading import local as tlocal
+    from threading import local as tlocal, Lock
 except ImportError:
-    from dummy_threading import local as tlocal
+    from dummy_threading import local as tlocal, Lock
+
+_tdata = tlocal()
```

Comment:
added comments
access to this object modifies the thread local data, kind of like thread locals in C
the old wrong usage created a new object instead of reusing a thread global one like I originally though (the docs are not too great)